### PR TITLE
Obituary creation

### PR DESCRIPTION
**Pull request by jpc20**
Added the ability for a user to create a new obituary.
Set up route for a logged-in user to click a button in the navbar and go through the new obituary form to the obituary show page to see the newly created obituary.
Added the one to many relationship between users and obituaries.
Added a sad path test for not filling out a name when creating an obituary, but still need to test that a visitor cant create one.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -15,6 +15,7 @@
         <form class="form-inline">
           <% if current_user %>
             <button class="btn btn-sm btn-outline-secondary" type="button">Logout</button>
+            <a href="/obituaries/new"><button class="btn btn-sm btn-outline-secondary" type="button">Add Obituary</button></a>
```
Comment (jpc20) [inline app/views/layouts/application.html.erb:18]: Anyone know of a better way to make a button a link?

File: app/views/obituaries/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<%= form_for @obituary do |f| %>
```
Comment (jpc20) [inline app/views/obituaries/_form.html.erb:1]: Added this form as a partial so we can reuse it in the edit form.

File: app/views/obituaries/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div class="obituary-image">
```
Comment (jpc20) [inline app/views/obituaries/show.html.erb:1]: This still needs to be styled. I just added the bare minimum to get it to pass the tests.

File: spec/factories/obituaries.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+  factory :obituary do
+    first_name { "MyString" }
+    last_name { "MyString" }
+    age { 1 }
+    city { "MyString" }
+    state { "MyString" }
+    description { "MyString" }
+    image_url { "MyString" }
```
Comment (jpc20) [inline spec/factories/obituaries.rb:9]: Still need to add faker to these.

File: app/controllers/obituaries_controller.rb [hunk 5]
```diff
@@ -1,4 +1,29 @@
...
+  end
+
+  def show
+    @obituary = Obituary.find(params[:id])
+  end
+
+  private
+
+  def obituary_params
+    params.require(:obituary).permit(:first_name, :last_name, :description, :age, :city, :state, :image_url)
```
Comment (sagemlee) [inline app/controllers/obituaries_controller.rb:27]: Looks good, I think these attributes will work for us in the mean time.

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (sagemlee) [reply app/views/layouts/application.html.erb:18]: I believe you can add the href part within the button tag, not usre if that would be any better.
`<button class="btn btn-sm btn-outline-secondary" type="button" href="/obituaries/new">Add Obituary</button>`

File: app/views/obituaries/_form.html.erb (hunk 2 above)
Comment (sagemlee) [reply app/views/obituaries/_form.html.erb:1]: Awesome, thank for doing that!

File: spec/factories/obituaries.rb (hunk 4 above)
Comment (sagemlee) [reply spec/factories/obituaries.rb:9]: @stellakunzang is really good at that :)

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (stellakunzang) [reply app/views/layouts/application.html.erb:18]: I looked in the bootstrap documentation and it said the same thing Sage wrote. I am wondering if you launch `rails s`, if the button works and the problem lies in getting the test to find/click the button.

File: app/views/obituaries/_form.html.erb (hunk 2 above)
Comment (stellakunzang) [reply app/views/obituaries/_form.html.erb:1]: Nice, this is awesome!

File: config/routes.rb [hunk 6]
```diff
@@ -3,14 +3,14 @@
...
 
-
    get '/profile', to: "users#show"
    resources :users, only: [:new, :create]
    get "/obituaries", to: "obituaries#index"
    namespace :obituaries do
      get 'covid-19', to: 'covid#index'
      get 'recent', to: 'recent#index'
    end
+   resources :obituaries, only: [:new, :create, :show]
```
Comment (stellakunzang) [inline config/routes.rb:13]: Do you want to combine this with the `get "/obituaries", to: "obituaries#index"` just to keep the routes tidy?

File: spec/factories/obituaries.rb (hunk 4 above)
Comment (stellakunzang) [reply spec/factories/obituaries.rb:9]: yay Faker!

File: spec/factories/user.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+FactoryBot.define do
+  factory :user do
+    email  { Faker::Internet.email }
+    first_name { Faker::Creature::Dog.name }
+    last_name { Faker::Artist.name }
+    password { Faker::Color.color_name }
```
Comment (stellakunzang) [inline spec/factories/user.rb:6]: this is a fun idea for faker passwords :)

File: config/routes.rb (hunk 6 above)
Comment (jpc20) [reply config/routes.rb:13]: Ooh yeah, that's a good idea. I'll make that change real quick.

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (jpc20) [reply app/views/layouts/application.html.erb:18]: Hmm yes, I think that was the problem. It wasn't showing up on the test. I'll try it out in `rails s` to see if that works

Comment (jpc20) [reply app/views/layouts/application.html.erb:18]: Actually, It still doesn't seem to work in `rails s` when the button is `<button class="btn btn-sm btn-outline-secondary" type="button" href="/obituaries/new">Add Obituary</button>` clicking it doesn't seem to do anything...

File: app/controllers/obituaries_controller.rb (hunk 5 above)
Comment (EricLarson2020) [reply app/controllers/obituaries_controller.rb:27]: Nice work here

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (EricLarson2020) [reply app/views/layouts/application.html.erb:18]: Thats the best methods I found, especially with get requests

File: app/views/obituaries/_form.html.erb (hunk 2 above)
Comment (EricLarson2020) [reply app/views/obituaries/_form.html.erb:1]: Nice work here

File: spec/factories/obituaries.rb (hunk 4 above)
Comment (EricLarson2020) [reply spec/factories/obituaries.rb:9]: Nice use here

**Pull request merged**